UefiPayloadPkg/Library/FdtParserLib/FdtParserLib.c

```diff
@@ -901,6 +932,22 @@ ParseDtb (
   for (Node = FdtNextNode (Fdt, 0, &Depth); Node >= 0; Node = FdtNextNode (Fdt, Node, &Depth)) {
     NodePtr = (FDT_NODE_HEADER *)((CONST CHAR8 *)Fdt + Node + Fdt32ToCpu (((FDT_HEADER *)Fdt)->OffsetDtStruct));
     DEBUG ((DEBUG_INFO, "\n   Node(%08x)  %a   Depth %x\n", Node, NodePtr->Name, Depth));
+
+    // Ensure we don't submit a child of reserved-memory as main memory block.
+    NodeType = CheckNodeType (NodePtr->Name, Depth);
+    // Already found reserved block.
+    if (ReservedMemoryDepth != 0xFF && Depth > ReservedMemoryDepth) {
+      DEBUG ((DEBUG_INFO, "Skipping reserved-memory block.\n"));
+      continue;
+    } else if (ReservedMemoryDepth != 0xFF && Depth <= ReservedMemoryDepth) {
+      ReservedMemoryDepth = 0xFF;
+    }
+    // Newly found reserved block.
+    if (NodeType == ReservedMemory) {
+      ReservedMemoryDepth = Depth;

```

Comment:
This line throw error: warning C4244: '=': conversion from 'INT32' to 'UINT8', possible loss of data.
and found other build errors with this PR when set -DHAND_OFF_FDT_ENABLE=TRUE, please double check it.

Reply:
Depth is defined as INT32, and ReservedMemoryDepth is defined as  UINT8 above. May change the type to keep them consistent or cast it.